order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -200,12 +316,135 @@ public void cancelItemSubscription(CancelSubscriptionRequest request, StreamObse
+                        MarketDataRequest marketDataRequest = MarketDataRequest.newBuilder()
+                                .setClientId(request.getClientId())
+                                .setGoodName(itemForSubscribing)
+                                .build();
+
+                        MarketServiceGrpc.newStub(marketChannel)
+                                .subscribeForItem(marketDataRequest, new StreamObserver<TickResponse>() {
+                                    @Override
+                                    public void onNext(TickResponse tickResponse) {
+                                        tickResponses.add(tickResponse);
+                                    }
+
+                                    @Override
+                                    public void onError(Throwable throwable) {
+
+                                    }
+
+                                    @Override
+                                    public void onCompleted() {
+                                        responseObserver.onCompleted();
+                                    }
+                                });
+                    }
+                })
+                .build().start());
+    }
+
+    private void startOrderBookService(OrderBookServiceGrpc.OrderBookServiceImplBase orderBookService, long sleepTime) {
+        new Thread(() -> {
+            try {
+                Thread.sleep(sleepTime);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverName).directExecutor().addService(orderBookService)
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+    }
+
+    private void startMarketServiceWithSubscribeForItemOverridden(int numberOfTickResponses, String serverNameMarket, long sleepTime) {
+        new Thread(() -> {
+            try {
+                Thread.sleep(sleepTime);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverNameMarket).directExecutor().addService(new MarketServiceGrpc.MarketServiceImplBase() {
+                            @Override
+                            public void subscribeForItem(MarketDataRequest request, StreamObserver<TickResponse> responseObserver) {
+
+                                for (int i = 0; i < numberOfTickResponses; i++) {
+                                    TickResponse tickResponse = TickResponse.newBuilder()
+                                            .setGoodName(request.getGoodName())
+                                            .setQuantity(i + productQuantity)
+                                            .setPrice(i + productPrice)
+                                            .setTimestamp(System.currentTimeMillis())
+                                            .setOrigin(Origin.forNumber((i % 4))).build();
```

Comment:
What will happen if we add 5'th origin afterwards, we will never test this variant unless we remember all the places in the code where we specify the exact number, but as it is not the purpose of the test to examine the behaviour of the tick responses, maybe we can leave it like this.

Reply:
Fixed